### PR TITLE
Add support for Event Semantic Conventions

### PR DESCRIPTION
**Pull request by Oberon00**
Event semantic conventions were designed with span events in mind but should be extendable to log events. See the new tests for how this can be useful to add a YAML definition for the exception semantic conventions in the spec.

This adds two new elements to the YAML definition:
* A new semantic convention type "event", with no additional fields
* A new additional field for "span" semantic conventions to reference associated event semantic conventions.
  The generator validates that the semantic convention exists and the new data is available to the code templates,
  but is not yet used in the markdown output.

It also updates the JSON schema to only allow span_kind and the new events field if the semantic convention type is not specified (defaults to span) or is "span". Unfortunately, we then lose the `"additionalProperties": false`, this seems to be a limitation of the JSON schema language, see https://stackoverflow.com/questions/22689900/json-schema-allof-with-additionalproperties#comment83207554_23001194

Full disclosure: This is the last feature for this tool we had developed internally at Dynatrace before (promise! 😄), so I'm starting off by submitting the status quo here in hopes our code bases back in sync (we already use the event semantic convention type internally).

### CODE REVIEW

File: semantic-conventions/src/opentelemetry/semconv/version.py [hunk 1]
```diff
@@ -12,4 +12,4 @@
 #   See the License for the specific language governing permissions and
 #   limitations under the License.
 
-__version__ = "0.4.1"
+__version__ = "0.5.0"
```
Comment (mariojonke) [inline semantic-conventions/src/opentelemetry/semconv/version.py:15]: i guess the [changelog](https://github.com/open-telemetry/build-tools/blob/main/semantic-conventions/CHANGELOG.md) should also be updated?

Comment (Oberon00) [reply semantic-conventions/src/opentelemetry/semconv/version.py:15]: Oh, I did not realize we had one, yes it should.

Comment (Oberon00) [reply semantic-conventions/src/opentelemetry/semconv/version.py:15]: 4720769

**Review by mariojonke — APPROVED** [review]

**Comment by Oberon00** [comment]
@open-telemetry/specs-approvers this still needs reviews.

**Comment by tigrannajaryan** [comment]
Maybe @open-telemetry/python-approvers can help review the Python portion and @open-telemetry/specs-logs-approvers can help review the event-related logic?

**Comment by Oberon00** [comment]
The events here were designed for span events. But if something prevents them from being used as logs (other than missing features) that would be good to know.

File: semantic-conventions/src/tests/data/yaml/errors/events/missing_event.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        brief: An attribute.
+
+  - id: random.event
+    type: event
+    prefix: random
+    brief: example
+    attributes:
+        - id: flag
+          type: boolean
+          brief: An attribute.
```
Comment (aabmass) [inline semantic-conventions/src/tests/data/yaml/errors/events/missing_event.yaml:21]: nit formatting

```suggestion
      - id: flag
        type: boolean
        brief: An attribute.
```

File: semantic-conventions/src/tests/data/yaml/errors/events/no_event_type.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        brief: An attribute.
+
+  - id: random.event
+    type: span
+    prefix: random
+    brief: example
+    attributes:
+        - id: flag
+          type: boolean
+          brief: An attribute.
```
Comment (aabmass) [inline semantic-conventions/src/tests/data/yaml/errors/events/no_event_type.yaml:20]: nit
```suggestion
      - id: flag
        type: boolean
        brief: An attribute.
```

File: semantic-conventions/src/tests/data/yaml/span_event.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        brief: An attribute.
+
+  - id: random.event
+    type: event
+    prefix: random
+    brief: example
+    attributes:
+        - id: flag
+          type: boolean
+          brief: An attribute.
```
Comment (aabmass) [inline semantic-conventions/src/tests/data/yaml/span_event.yaml:21]:
```suggestion
      - id: flag
        type: boolean
        brief: An attribute.
```

**Review by aabmass — APPROVED** [review]
LGTM

File: semantic-conventions/src/tests/data/yaml/errors/events/no_event_type.yaml (hunk 3 above)
Comment (Oberon00) [reply semantic-conventions/src/tests/data/yaml/errors/events/no_event_type.yaml:20]: Thanks! Maybe we should add yamllint to this repo too.

**Review by thisthat — APPROVED** [review]

**Pull request merged**